### PR TITLE
allow other pkgs to check for blobs in pruning cache

### PR DESCRIPTION
**Pull request by kasey**
**What type of PR is this?**

Incremental improvement

**What does this PR do? Why is it needed?**

This adds to the filesystem package the `BlobStorageSummarizer` interface, which can be used to efficiently check whether a given root+slot is available on disk. Under the hood this allows other packages to access the blob pruner cache, which builds and maintains a consistent of the filesystem via a cache warmup method that crawls all blob directories at startup, notification of all new blob writes.

Since this view isn't consistent until after the cache has been warmed, the summarizer is obtained by calling `BlobStorage.WaitForSummarizer`, which will block until the cache is ready to be used.

This feature will be used in an upcoming PR to improve initial-sync bandwidth usage by skipping slots in the block-by-range request, when we already have the corresponding blobs on disk.

### CODE REVIEW

File: beacon-chain/db/filesystem/cache.go [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+	"github.com/prysmaticlabs/prysm/v5/config/params"
+	"github.com/prysmaticlabs/prysm/v5/consensus-types/primitives"
+)
+
+// blobIndexMask is a bitmask representing the set of blob indices that are currently set.
+type blobIndexMask [fieldparams.MaxBlobsPerBlock]bool
+
+// BlobStorageSummary represents cached information about the BlobSidecars on disk for each root the cache knows about.
+type BlobStorageSummary struct {
+	Slot primitives.Slot
```
Comment (saolyn) [inline beacon-chain/db/filesystem/cache.go:16]: Does slot need to be exported?

File: beacon-chain/db/filesystem/cache.go [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+func (s BlobStorageSummary) HasIndex(idx uint64) bool {
+	// Protect from panic, but assume callers are sophisticated enough to not need an error telling them they have an invalid idx.
+	if idx >= fieldparams.MaxBlobsPerBlock {
+		return false
+	}
+	return s.mask[idx]
+}
+
+// AllAvailable returns true if we have all blobs for all indices from 0 to count-1.
+func (s BlobStorageSummary) AllAvailable(count int) bool {
```
Comment (saolyn) [inline beacon-chain/db/filesystem/cache.go:30]: Could you add a unit test for this function?

File: beacon-chain/db/filesystem/cache.go (hunk 1 above)
Comment (kasey) [reply beacon-chain/db/filesystem/cache.go:16]: no, will make it private

Comment (kasey) [reply beacon-chain/db/filesystem/cache.go]: done

File: beacon-chain/db/filesystem/cache.go (hunk 2 above)
Comment (kasey) [reply beacon-chain/db/filesystem/cache.go:30]: done - found an off-by-one in the process :+1:

File: beacon-chain/db/filesystem/cache_test.go [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+	}
+	for i := 0; i < count; i++ {
+		if !s.mask[i] {
+			return false
+		}
+	}
+	return true
+}
+
+*/
```
Comment (saolyn) [inline beacon-chain/db/filesystem/cache_test.go:85]: Is this comment supposed to be here?

**Review by terencechain — APPROVED** [review]

**Pull request merged**